test/integration/client/cancel-query-tests.js

```diff
@@ -1,5 +1,20 @@
 var helper = require(__dirname+"/test-helper");
 
+test("cancellation of an active query", function() {
+  var client = helper.client();
+
+  var qry = "select pg_sleep(4)";
+  var query1 = client.query(qry);
+  client.cancel(client, query1);
+  var otherClient = helper.client();
+  activeQueries = "SELECT * FROM pg_stat_activity WHERE query LIKE '%pg_sleep(4)%' AND query NOT LIKE '%pg_stat_activity%'";
+  otherClient.query(activeQueries, assert.calls(function(err, result) {
+    assert.equal(result.rows.length, 0);
+    otherClient.end();
+    client.end();
```

Comment:
The process wouldn't exit without these lines; do you know why? (Obviously it's good to clean up resources, it just seemed kind of like magic)